tests/unit/analytics/test_views.py

```diff
@@ -429,35 +429,33 @@ def test_get_challenge_phase_submission_analysis_when_challenge_phase_does_not_e
         self.assertEqual(response.data, expected)
         self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
 
-    def test_get_challenge_phase_submission_analysis_for_participant_team_1(self):
+    def test_get_challenge_phase_submission_count_by_team_for_participant_team_1(self):
         self.challenge.participant_teams.add(self.participant_team)
         self.challenge.participant_teams.add(self.participant_team3)
 
-        self.url = reverse_lazy('analytics:get_challenge_phase_submission_analysis',
+        self.url = reverse_lazy('analytics:get_challenge_phase_submission_count_by_team',
                                 kwargs={'challenge_pk': self.challenge.pk,
                                         'challenge_phase_pk': self.challenge_phase.pk})
 
         expected = {
-                "submission_count": 3,
-                "participant_team_count": 2,
+                "participant_team_submission_count": 3,
```

Comment:
The number `3` should not be hardcoded here. It should be fetched from the model. Something like `participant_team_1.submissions.count()`

Reply:
Hmm..I did this intentionally so that it will be easy in future to get a sense of the test case as there are two test cases one in which `participant_team_submission_count` is 3 and other one in which `participant_team_submission_count` is 1. Should I change it back?

Reply:
The reason why I am saying to remove hard coding is that let's say sometime in future we change the number of submissions done by a particular team in the tests, then your all tests will break for no reason which should not happen.

Reply:
yes! Agreed. 👍